.github/workflows/phpstan.yaml

```
@@ -0,0 +1,30 @@
+name: Static analysis - phpstan
+
+on:
+  pull_request:
+    paths:
+      - '**.php'
+
+jobs:
+  phpstan:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '8.0'
+          extensions: "json,memcached"
+          ini-values: "memory_limit=-1"
+          coverage: none
+
+      - name: Checkout target branch
+        uses: actions/checkout@v2
+
+      - name: 'Install project dependencies'
+        run: |
+          composer update --no-interaction --prefer-dist --optimize-autoloader --prefer-stable
+
+      - name: phpstan
+        run: |
+          ./vendor/bin/phpstan --no-interaction --no-progress
```

You should precise a level here with a `--level` https://phpstan.org/user-guide/rule-levels
By default it's 0.

```suggestion
          ./vendor/bin/phpstan --no-interaction --no-progress --level=0
```
IMO It's easier when you want to increase the level to have it explicitly in the pipeline